esm/html/document.js

```diff
@@ -102,3 +103,53 @@ export class HTMLDocument extends Document {
     return element;
   }
 }
+
+/** @typedef {{ [WINDOW] : Object }} Context */
+/**
+ * @constructor
+ * @param {Context} context
+ */
+export class HTMLDocument extends _HTMLDocument {
```

Comment:
There is no need to keep _HTMLDocument and HTMLDocument separate, but I thought it would be clearer to understand this PR this way.

Reply:
I think any document should accept extras, not just the HTML kind

Reply:
Okay, I agree that seems like a good idea.